eZ/Publish/Core/FieldType/User/SearchField.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\Core\FieldType\User;
+
+use eZ\Publish\SPI\FieldType\Indexable;
+use eZ\Publish\SPI\Persistence\Content\Field;
+use eZ\Publish\SPI\Persistence\Content\Type\FieldDefinition;
+use eZ\Publish\SPI\Search;
+
+final class SearchField implements Indexable
+{
+    public function getIndexData(Field $field, FieldDefinition $fieldDefinition): array
+    {
+        return [
+            new Search\Field(
+                'fulltext_email',
+                $field->value->externalData['email'],
+                new Search\FieldType\FullTextField()
```

Comment:
Hi, isn't exposing user emails in fulltext search a kind of security/privacy risk?

Reply:
Hi @pspanj, good input, afaik security will be dealt with by the limitations applied to query. So this is only a issue if people search with security filtering turned off afaik. Which is then exposing any content.

Or where you thinking of something else?

Also someone mentioned this is kind of how it worked in legacy, but TODO is to verify that for sure.

Reply:
Hi. @pspanja besides what André just have mentioned, this is opt-in. By default "is searchable" is disabled for existing `ezuser` Field Definitions. We're looking into how to improve doc and UI so an Administrator could be more aware of possible consequences.

Reply:
Hi guys, that's clear, but it still does not give you enough control. For example, if you allow read of User Content to anonymous role, they will automatically get fulltext search on emails if the User field is marked as searchable, which is something you might not want. @andrerom if you remember we discussed this some time ago, even if User is a Content, it is also another domain, and we don't distinguish this from the search API. But that's probably a separate topic.

From my POV this is something specific enough that the need for custom Indexable implementation for the User field type would be acceptable.

BTW I wonder what is the use case for fulltext search on emails? Maybe a separate criterion for searching on email would be an acceptable alternative?

Reply:
> BTW I wonder what is the use case for fulltext search on emails?

Hi. It's about the expectation that user can be found using Search in AdminUI, which uses FullText criterion. FullText also serves another purpose - you can specify "@domain.com" to find e-mail.

> it still does not give you enough control.

Alternatively indeed we could model that as "wildcard" Criterions: `Criterion\User\Email` and maybe `Criterion\User\Login`, because in that case it makes sense. It would have to be `%like%` based in LSE, so "@domain.com" also works. We could cover other use-cases which right now are a bit of a challenge, especially for LSE, like matching "@domain.com", but not "@domain.com.pl".
This of course implies much more work for AdminUI, because filtering needs to be adjusted to use that Criterion for user filter.
I think @adamwojs suggested something similar internally btw ;)

cc @andrerom @kmadejski

Reply:
In than case take a look at this, and other 4 user criteria on the same level: https://github.com/netgen/ezplatform-search-extra/blob/master/lib/API/Values/Content/Query/Criterion/UserEmail.php

Reply:
> Alternatively indeed we could model that as "wildcard" Criterions: Criterion\User\Email and maybe Criterion\User\Login, because in that case it makes sense. It would have to be %like% based in LSE, so "@domain.com" also works. We could cover other use-cases which right now are a bit of a challenge, especially for LSE, like matching "@domain.com", but not "@domain.com.pl".
This of course implies much more work for AdminUI, because filtering needs to be adjusted to use that Criterion for user filter.
I think @adamwojs suggested something similar internally btw ;)

Dedicated user-related criteria is a good idea: from security and also DX perspective.